scripts/jupyterkernel/kernelsds/src/main/java/com/kernelsds/app/ISystemDsKernel.java

```diff
@@ -0,0 +1,399 @@
+            .preferLong()
+            //Keywords from a great poem at https://stackoverflow.com/a/12114140
+            .withKeywords("let", "this", "long", "package", "float")
+            .withKeywords("goto", "private", "class", "if", "short")
+            .withKeywords("while", "protected", "with", "debugger", "case")
+            .withKeywords("continue", "volatile", "interface")
+            .withKeywords("instanceof", "super", "synchronized", "throw")
+            .withKeywords("extends", "final", "export", "throws")
+            .withKeywords("try", "import", "double", "enum")
+            .withKeywords("false", "boolean", "abstract", "function")
+            .withKeywords("implements", "typeof", "transient", "break")
+            .withKeywords("void", "static", "default", "do")
+            .withKeywords("switch", "int", "native", "new")
+            .withKeywords("else", "delete", "null", "public", "var")
+            .withKeywords("in", "return", "for", "const", "true", "char")
+            .withKeywords("finally", "catch", "byte")
+            .build();
+
+    private static final CharPredicate idChar = CharPredicate.builder()
+            .inRange('a', 'z')
+            .inRange('A', 'Z')
+            .match('_')
+            .build();
+
+
+    private LanguageInfo languageInfo;
+    private final Connection connection = new Connection();
+    private int sca;
+    private HashMap<String, Object> hashMap = new HashMap<>();
+    private HashMap<String, double[][]> hashMapForMatrix = new HashMap<>();
+
+    public ISystemDsKernel() {
+
+    }
+
+
+
+    @Override
+    public LanguageInfo getLanguageInfo() {
+        return languageInfo;
+    }
+
+
+    /*String script =
+                "  X = rand(rows=10, cols=10);"
+                        + "R = matrix(0, rows=10, cols=1)"
+                        + "parfor(i in 1:nrow(X))"
+                        + "  R[i,] = sum(X[i,])"
+                        + "print(sum(R))";*/
+
+    public double[][] stringToDoubleArray(String matrixString) {
+        // Split the input string into rows.
+        String[] rows = matrixString.split(";");
+
+        // Prepare the resulting double array with the correct number of rows.
+        double[][] matrix = new double[rows.length][];
+
+        for (int i = 0; i < rows.length; i++) {
+            // Split each row into its individual values.
+            String[] values = rows[i].split(",");
+
+            // Prepare the row array.
+            matrix[i] = new double[values.length];
+
+            for (int j = 0; j < values.length; j++) {
+                // Convert each value to double and store in the matrix.
+                try {
+                    matrix[i][j] = Double.parseDouble(values[j].trim()); // Trim to remove any leading or trailing spaces.
+                } catch (NumberFormatException e) {
+                    // Handle the case where the string cannot be parsed as a double.
```

Comment:
These parsing logic seem to make strong assumptions regarding the code statements. I like to see some examples of complex dml code. You can use our builtins.